ddtrace/llmobs/_llmobs.py

```diff
@@ -1704,98 +1724,222 @@ def submit_evaluation(
+        :param str ml_app: The name of the ML application for the span(s) to run evaluations on.
+        :param str from_timestamp: The unix timestamp in milliseconds representing the beginning of the
+            time range to filter spans by.
+        :param str to_timestamp: The unix timestamp in milliseconds representing the end of the time range to
+            filter spans by.
+        :param list[Callable[[Dict[str, Any]], LLMObsEvaluationResult]] evaluations: A list of evaluation
+            functions to run on the spans that match the given filters.
+            Each function should accept a dictionary representing an exported span and return an LLMObsEvaluationResult.
+            See https://docs.datadoghq.com/llm_observability/evaluations/export_api/?tab=model#searchedspan for
+            details on the exported span format.
+        """
+        if not evaluations:
+            return
 
-            evaluation_metric: LLMObsEvaluationMetricEvent = {
-                "join_on": join_on,
-                "label": str(label),
-                "metric_type": metric_type,
-                "timestamp_ms": timestamp_ms,
-                "{}_value".format(metric_type): value,  # type: ignore
-                "ml_app": ml_app,
-                "tags": ["{}:{}".format(k, v) for k, v in evaluation_tags.items()],
-            }
+        for exported_span in cls._instance._export_spans_client.export_spans(
+            span_id=span_id,
+            trace_id=trace_id,
+            tags=tags,
+            span_kind=span_kind,
+            span_name=span_name,
+            ml_app=ml_app,
+            from_timestamp=from_timestamp,
+            to_timestamp=to_timestamp,
+        ):
+            for evaluation in evaluations:
+                evaluation_result: LLMObsEvaluationResult = {}
+                error = None
+                join_on = {
+                    "span": {
+                        "span_id": exported_span.get("span_id"),
+                        "trace_id": exported_span.get("trace_id"),
+                    },
+                }
+                metric_type = ""
+                try:
+                    evaluation_result = evaluation(exported_span)
+                    evaluation_metric = cls._build_evaluation_metric(evaluation_result, join_on, exported_span)
+                    cls._instance._llmobs_eval_metric_writer.enqueue(evaluation_metric)
+                except LLMObsSubmitEvaluationError as e:
+                    error = e.error_type
+                    log.error(
+                        "Failed to submit evaluation metric %s for span %s",
+                        evaluation_result.get("label"),
+                        exported_span.get("span_id"),
+                    )
+                finally:
+                    metric_type = evaluation_result.get("metric_type") or ""
+                    telemetry.record_llmobs_submit_evaluation(join_on, metric_type, error)
```

Comment:
a thought - maybe we wanna have an extra tag for this submit evaluation telemetry signal that it happened through the `run_evaluations` method, maybe like `source:run_evaluations`. if you also think that's a good idea, we'd have to update the metric definition in the backend to accept a new tag, although if we're worried about cardinality for that metric we could make a new one.

either way i think it might be nice to have a signal of how many people are using this `run_evaluations` method vs the normal `submit_evaluations` one